lib/index.ts

```diff
@@ -0,0 +1,244 @@
+// Copyright 2016 Google Inc. All Rights Reserved.
+// Licensed under the Apache License, Version 2.0. See LICENSE
+
+const lighthouse = require('lighthouse');
+const ChromeLauncher = require('lighthouse/lighthouse-cli/chrome-launcher');
+const perfConfig: any = require('lighthouse/lighthouse-core/config/perf.json');
+const opn = require('opn');
+
+const Sheets = require('./sheets/index');
+const Chart = require('./chart/chart');
+const metrics = require('./metrics');
+const expectations = require('./expectations');
+const {upload} = require('./upload');
+const messages = require('./utils/messages');
+
+import {MainOptions, FeatureFlags, AuthorizeCredentials, LighthouseResults, MetricsResults, DriveResponse, TermWritableStream} from '../types/types';
+
+const _MAX_LIGHTHOUSE_TRIES = 2;
+const _SIGINT = 'SIGINT';
+const _SIGINT_EXIT_CODE = 130;
+
+class PWMetrics {
+  flags: FeatureFlags|any;
+  runs: number;
+  sheets: Object;
+  expectations: Object;
+  clientSecret: AuthorizeCredentials;
+  tryLighthouseCounter: number;
+  launcher: any;
+
+  constructor(public url: string, opts: MainOptions) {
+    this.flags = opts.flags || {};
+    // @todo remove when new lighthouse version be released, because -https://github.com/GoogleChrome/lighthouse/pull/1778
+    this.flags.disableCpuThrottling = this.flags.disableCpuThrottling || false;
+    this.runs = this.flags.runs || 1;
+    this.sheets = opts.sheets;
```

Comment:
No default value? Yargs does not have an default value for this either so it may be good for input validation to provide some fallback.